components/dashboard/src/components/WorkspaceClassesOptions.tsx

```diff
@@ -0,0 +1,202 @@
+import { AllowedWorkspaceClass, DEFAULT_WS_CLASS } from "../data/workspaces/workspace-classes-query";
+import { MiddleDot } from "./typography/MiddleDot";
+import { useToast } from "./toasts/Toasts";
+import Modal, { ModalBody, ModalFlexFooter, ModalFooterAlert, ModalHeader } from "./Modal";
+
+export const WorkspaceClassesOptions = (props: {
+    classes: AllowedWorkspaceClass[];
+    defaultClass?: string;
+    className?: string;
+}) => {
+    return (
+        <div className={cn("space-y-2", props.className)}>
+            {props.classes.map((cls) => (
+                <div className="flex gap-2 items-center">
+                    <CpuIcon size={20} />
+                    <div>
+                        <span className="font-medium text-pk-content-primary">{cls.displayName}</span>
+                        <MiddleDot />
+                        <span className="text-pk-content-primary">{cls.description}</span>
+                        {props.defaultClass === cls.id && (
+                            <>
+                                <MiddleDot className="text-pk-content-tertiary" />
+                                <span className="text-pk-content-tertiary">default</span>
+                            </>
+                        )}
+                    </div>
+                </div>
+            ))}
+        </div>
+    );
+};
+
+export interface WorkspaceClassesModifyModalProps {
+    defaultClass?: string;
+    restrictedWorkspaceClasses: string[];
+    showSetDefaultButton: boolean;
+
+    allowedClasses: AllowedWorkspaceClass[];
+    updateMutation: UseMutationResult<void, Error, { restrictedWorkspaceClasses: string[]; defaultClass?: string }>;
+
+    onClose: () => void;
+}
+
+export const WorkspaceClassesModifyModal = ({
+    onClose,
+    updateMutation,
+    allowedClasses,
+    showSetDefaultButton,
+    ...props
+}: WorkspaceClassesModifyModalProps) => {
+    const [defaultClass, setDefaultClass] = useState(props.defaultClass || DEFAULT_WS_CLASS);
+    const [restrictedClasses, setRestrictedClasses] = useState(props.restrictedWorkspaceClasses ?? []);
+    const { toast } = useToast();
+
+    const handleUpdate = async () => {
+        if (computedError) {
+            return;
+        }
+        updateMutation.mutate(
+            { restrictedWorkspaceClasses: restrictedClasses, defaultClass },
+            {
+                onSuccess: () => {
+                    toast({ message: "Workspace class updated" });
+                    onClose();
+                },
+            },
+        );
+    };
+
+    const computedError = useMemo(() => {
+        const leftOptions =
+            allowedClasses.filter((c) => !c.isDisabledInScope && !restrictedClasses.includes(c.id)) ?? [];
+        if (leftOptions.length === 0) {
+            return "Must have at least one class.";
+        }
+        if (!defaultClass || !leftOptions.find((cls) => cls.id === defaultClass)) {
+            return "Must have default class.";
```

Comment:
```suggestion
            return "A default workspace class is required.";
```

Reply:
Too many words will make Modal weird. So I tried to make it as short as possible.

@corygrunk could we somehow make Modal wider? I will update this comment later with screenshot

Reply:
👍  Checked `A default workspace class is required.` is working with current width (not small window, small windows all have this problem, so ignore it first).

For classes switch (es), error message is too long `At least one workspace class has to be selected.` if we want to have it like the one we use in org-level

https://github.com/gitpod-io/gitpod/blob/622c3529ee9273ce62a3f7439d74b46d8b3fef2b/components/dashboard/src/teams/TeamSettings.tsx#L477

Reply:
With two lines is not that bad?
<img width="969" alt="image" src="https://github.com/gitpod-io/gitpod/assets/20944364/2ea3738e-2623-42df-889a-921af2fe8346">

Reply:
@corygrunk  For error messages from requests, request to save settings might be failed (i.e. Network issue or something else), should we show that kind of error at this place? (it may show a long string)

In other Modal components, we will show a Alert with animation (see image below)

|Image|
|:-:|
|<img width="619" alt="image" src="https://github.com/gitpod-io/gitpod/assets/20944364/5117a5a3-619e-4224-bab8-b63158d92dda">|

Reply:
If it's too long, we can also shorten it given some context cues, like:
```
A default is required.
```

Reply:
Choose to go with show whatever error message next to buttons

|Image|
|:-:|
|<img width="857" alt="image" src="https://github.com/gitpod-io/gitpod/assets/20944364/a2c2410c-a737-4ca4-8094-7374aa101457">|